### DISCUSSION TITLE
Flows "Run script" operation: support import or clarify documentation

### DISCUSSION THREAD

**Discussion by HugoGresse** [directus/directus#16886] [Archive]
Following [Operation#run-script](https://docs.directus.io/configuration/flows/operations.html#run-script) documentation, this is not explicit of what is allowed. 

For example: we are not able to do `import` or `require` for a standard nodejs library (https or crypto). 

Using Crypto is flows is very useful to verify data and generate token for example. 

It could either be great to clarify the Flow run script doc or link to the source defining what is allow in run script. 

EDIT: source define additional module allowed via the env: https://github.com/directus/directus/blob/main/api/src/operations/exec/index.ts

**Comment by rijkvanzanten** [comment]
Added a note to the docs to clarify: https://github.com/directus/docs/commit/10d1c6529f713d3e8be87c35d545f814ac1ec5f2

All node module use (system or otherwise) is blocked by default for security reasons. Individual modules can be allowlisted with the linked env var 👍🏻

**Reply by HugoGresse** [reply]
Perfect thanks! I'll try to contribute to the docs project directly, I think there are many small missing parts to be more explicit.

**Reply by rijkvanzanten** [reply]
> Perfect thanks! I'll try to contribute to the docs project directly, I think there are many small missing parts to be more explicit. 

🙏🏻 for sure! It's hard to spot little gabs like this when looking at the big picture :)

**Reply by Nisthar** [reply]
@rijkvanzanten can you tell me what is the security risk? only the admin can set flows Right?

**Reply by rijkvanzanten** [reply]
> @rijkvanzanten can you tell me what is the security risk? only the admin can set flows Right?

An admin in the directus api still has lower permission than a system admin who has access to the machine directus is running on. Node has full access to the machine, so by leaving all modules available by default, we'd open up a huge attack vector if you end up with elevated permissions within directus.
